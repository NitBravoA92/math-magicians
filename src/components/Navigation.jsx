import NavigationLink from './NavigationLink';
import routesList from '../routes/routesList';

const Navigation = () => {
  
  return (
    <nav>
      <ul id="menu">
        {
          routesList.map((link, index) => <NavigationLink { ...link } key={index + 1} />)
        }
      </ul>
    </nav>
  );
}

export default Navigation;