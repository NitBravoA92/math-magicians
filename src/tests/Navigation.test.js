import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Navigation from '../components/Navigation';

describe('The Navigation component', () => {
  test('renders correctly into the DOM', () => {
    const component = renderer
      .create(<Navigation />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  test('should render 3 links with the titles: Home, Calculator, Quote', () => {
    const component = render(<Navigation />);

    const links = component.container.querySelectorAll('a');

    expect(links.length).toBe(3);

    expect(component.container).toHaveTextContent('Home');
    expect(component.container).toHaveTextContent('Calculator');
    expect(component.container).toHaveTextContent('Quote');
  });
});
