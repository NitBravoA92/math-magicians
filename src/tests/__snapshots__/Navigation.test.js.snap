// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Navigation component renders correctly into the DOM 1`] = `
<nav>
  <ul
    id="menu"
  >
    <li
      className="nav-item"
    >
      <a
        href="/"
      >
        Home
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        href="/calculator"
      >
        Calculator
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        href="/quote"
      >
        Quote
      </a>
    </li>
  </ul>
</nav>
`;
